compiler/src/main/java/com/squareup/anvil/compiler/Utils.kt

```diff
@@ -136,7 +136,9 @@ internal fun AnnotationDescriptor.getAnnotationValue(key: String): ConstantValue
   allValueArguments[Name.identifier(key)]
 
 internal fun AnnotationDescriptor.scope(module: ModuleDescriptor): ClassDescriptor {
-  val kClassValue = requireNotNull(getAnnotationValue("scope")) as KClassValue
+  val kClassValue = requireNotNull(getAnnotationValue("scope")) {
+    "Couldn't find scope for ${module.name}"
```

Comment:
I suggest to add a test for that. I don't believe `module.name` is what you expect. It's also weird that this is needed in the first place. Declaring the scope is required, otherwise there's a compile error:

![image](https://user-images.githubusercontent.com/1558045/109720024-61082c80-7b5e-11eb-89ab-cbfe2ee52c2a.png)

```
e: com.squareup.anvil.compiler.AnvilCompilationException: Back-end (JVM) Internal error: The first argument for com.squareup.anvil.annotations.ContributesTo must be a class literal: @Module
@ContributesTo
public object AppModule2 {
  @Provides @Singleton public fun provideFunction(): (String) -> Int = { it.length }
  @Provides internal fun provideType(): CharSequence = "Hello"
}
File being compiled: (12,1) in /Users/ralf/Development/projects/square/anvil/integration-tests/library/src/main/java/com/squareup/anvil/test/Modules.kt
```

I guess Anvil runs before normal Kotlin compilation and then this error is thrown. A test will help to find out where the ambiguous error is thrown. I don't know in which exact scenario, but once you add the test I can point you to the best location.

Reply:
Yeah, I think that is the scenario I'm seeing, anvil fails before kotlin runs, and it was on a big rebase, so I had no idea where to look. I'll work on a test.

Reply:
Thanks. There are a few examples already. I also suggest to use `AnvilCompilationException` rather than `requireNotNull`, because then you get better error messages. I was lazy in this code snippet in the past 🙃